pages/web-literacy/WebLiteracyPage.jsx

```diff
@@ -1,44 +1,314 @@
 var React = require('react');
 var OutboundLink = require('react-ga').OutboundLink;
+var ReactRouter = require('react-router');
+var History = ReactRouter.History;
+var Link = ReactRouter.Link;
 
 var Illustration = require('../../components/illustration.jsx');
 
 var webmaps = require('./webmaplisting.jsx');
 
-var WebLiteracyPage = React.createClass({
+var CircleTree = require('react-circletree/es5');
+var weblitdataroot = require('./weblitdata');
+var weblitdata = weblitdataroot["WEB LITERACY"];
+var weblitcontent = require('./weblitcontent');
+var activitydata = require('./activitydata');
+var categories = require('./categories');
+var weblitcolors = require('./colors');
+var topicContent = weblitcontent.topics;
+var webLitSkillsContent = weblitcontent.webLitSkills;
+var makeLinksFromWebLitSkills = require("./MakeLinksFromWebLitSkills.jsx");
+
+function makeLinksFrom21CSkills(skills21C) {
+  return skills21C.map(function(skill21C, index) {
+    return (
+      <span key={skill21C} className="comma-separated-links">
+        <a href={"/web-literacy/skills/#" + categories[skill21C]}>
+          {categories[skill21C]}
+        </a>
+      </span>
+    );
+  });
+} 
+
+var Activity = React.createClass({
+  render: function() {
+    return (
+      <Illustration
+        width={200} height={200}
+        src1x={this.props.src1x}
+        src2x={this.props.src2x}
+        link={this.props.href}
+        externalLink={true}
+        alt={this.props.name + "-activity-icon"}>
+        <h2><a href={this.props.href}>{this.props.name}</a></h2>
+        <div className="meta-container">
+          <span className="meta-item"><i className="fa fa-users"></i>{this.props.difficulty}</span>
+          <span className="meta-item"><i className="fa fa-clock-o"></i>{this.props.duration}</span>
+        </div>
+        <p>{this.props.content}</p>
+        <div><b>Web Literacy Skills:</b> {makeLinksFromWebLitSkills(this.props.webLitSkills)}</div>
+        <div><b>21C Skills:</b> {makeLinksFrom21CSkills(this.props.skills)}</div>
+      </Illustration>
+    );
+  }
+});
+
+var WebLitPage = React.createClass({
+  mixins: [History],
   statics: {
     pageTitle: "Web Literacy",
     pageClassName: "web-literacy"
   },
-  render: function() {
-    var whitepaperLink = "https://mozilla.github.io/webmaker-whitepaper";
-    return (
-      <div className="inner-container">
-        <h1>Web Literacy</h1>
+  updateMapNavState: function() {
+    var topic = this.props.params.verb || "";
+    var webLitSkill = this.props.params.webLitSkill || "";
+    var state = {};
+    if (this.state.topic !== topic || this.state.webLitSkill !== webLitSkill) {
+      state.topic = topic;
+      state.webLitSkill = webLitSkill;
+      this.setState(state);
+    }
```

Comment:
Why are we not just doing this in `componentWillUpdate` or `componentDidUpdate`? Both give you the current props/state and next/previous (respectively) props/state pair to work with